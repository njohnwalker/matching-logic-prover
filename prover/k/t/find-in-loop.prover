claim \implies( \and( lsegright( variable("H0"), variable("OLDX"), variable("X"), variable("F1"), .Patterns)
                    , list(variable("H0"), variable("X"), variable("F2"), .Patterns)
                    , disjoint(variable("F1"), variable("F2"))
                    , \not(isMember(variable("DATA"), variable("F1")))
                    , gt(variable("X"), 0)
                    , gt(variable("X"), variable("DATA"))
                    , \equals(variable("X2"), select(variable("H0"), plus(variable("X"), 1)))
                    , \equals(variable("F3"), add(variable("F1"), variable("X")))
                    , \equals(variable("F4"), del(variable("F2"), variable("X")))
                    , .Patterns
                    )
              , \and( lsegright( variable("H0"), variable("OLDX"), variable("X2"), variable("F3"), .Patterns)
                    , list(variable("H0"), variable("X2"), variable("F4"), .Patterns)
                    , disjoint(variable("F3"), variable("F4"))
                    , \not(isMember(variable("DATA"), variable("F3")))
                    , .Patterns )
              )

strategy right-unfold-Nth(0,0) ; direct-proof

/*
strategy kt(lsegright) ; ( ( simplify ; direct-proof )
                        )
*/
