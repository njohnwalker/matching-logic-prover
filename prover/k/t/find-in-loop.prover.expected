<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( list ( variable ( "H0" ) , variable ( "X" ) , variable ( "F0" ) , .Patterns ) , \equals ( variable ( "X" ) , variable ( "OLDX" ) ) , \equals ( variable ( "TMP" ) , 0 ) , .Patterns ) , \and ( disjoint ( variable ( "F1" ) , variable ( "F2" ) ) , \not ( isMember ( variable ( "DATA" ) , variable ( "F1" ) ) ) , \equals ( variable ( "OLDX" ) , variable ( "X" ) ) , \equals ( variable ( "F1" ) , emptyset ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
         direct-proof
      ~> right-unfold-oneBody ( lsegright ( variable ( "H0" ) , variable ( "OLDX" ) , variable ( "X" ) , variable ( "F1" ) , .Patterns ) , \and ( \equals ( variable ( "OLDX" ) , variable ( "X" ) ) , \equals ( variable ( "F1" ) , emptyset ) , .Patterns ) )
    </trace>
  </goal>
</prover>
